### PR TITLE
Remove install_yamls dependency for kuttl tests

### PR DESCRIPTION
**Pull request by cescgina**
Instead of using install_yamls targets for deploying and cleaning up
keystone, use the CR sample from config/samples. This removes a possible
circular dependency issue when using install_yamls to run the jobs and
to run some test steps.

### CODE REVIEW

File: tests/kuttl/tests/change_keystone_config/00-deploy_keystone.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+../../common/keystoneapi_deploy.yaml
```
Comment (stuggi) [inline tests/kuttl/tests/change_keystone_config/00-deploy_keystone.yaml:1]: should there always be cleanup be triggered as 00 to make sure we start with fresh sample deployment?

**Review by stuggi — COMMENTED** [review]
we should probably rename the install_yaml `keystone_deploy_validate` target to something else, like keystone_kuttl as we no longer validate the deployment created with keystone_deploy ?

File: tests/kuttl/tests/change_keystone_config/00-deploy_keystone.yaml (hunk 1 above)
Comment (cescgina) [reply tests/kuttl/tests/change_keystone_config/00-deploy_keystone.yaml]: Done

File: tests/kuttl/common/keystoneapi_deploy.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../config/samples/keystone_v1beta1_keystoneapi.yaml
```
Comment (marios) [inline tests/kuttl/common/keystoneapi_deploy.yaml:1]: interesting... so we don't need to issue any oc commands... like kuttl will just apply the crd we have defined there? https://github.com/openstack-k8s-operators/keystone-operator/blob/b9019b4533f4049d41677329134d6dddb0ab0a63/config/samples/keystone_v1beta1_keystoneapi.yaml#L1

Comment (cescgina) [reply tests/kuttl/common/keystoneapi_deploy.yaml:1]: Exactly, if we pass a yaml file with kubernetes resources, kuttl will apply them to the cluster

**Comment by abays** [comment]
This PR should also update `kuttl-test.yaml`'s comment [1] on how to run the KUTTL tests.

[1] https://github.com/openstack-k8s-operators/keystone-operator/blob/0c0446d0c30868b0698a0e31dffc02de6f536e16/kuttl-test.yaml#L4

**Comment by cescgina** [comment]
> This PR should also update `kuttl-test.yaml`'s comment [1] on how to run the KUTTL tests.
> 
> [1]
> 
> https://github.com/openstack-k8s-operators/keystone-operator/blob/0c0446d0c30868b0698a0e31dffc02de6f536e16/kuttl-test.yaml#L2-L5

I've updated the comment since it was outdated and the target name changed. To clarify, however, we will still run the kuttl tests from install_yamls (at least in CI).

**Comment by abays** [comment]
> > This PR should also update `kuttl-test.yaml`'s comment [1] on how to run the KUTTL tests.
> > [1]
> > https://github.com/openstack-k8s-operators/keystone-operator/blob/0c0446d0c30868b0698a0e31dffc02de6f536e16/kuttl-test.yaml#L2-L5
> 
> I've updated the comment since it was outdated and the target name changed. To clarify, however, we will still run the kuttl tests from install_yamls (at least in CI).

Ah, thank you for clarifying.

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**